### PR TITLE
Seller controller service

### PR DESCRIPTION
**Pull request by tavisca-acharpe**

### CODE REVIEW

**Review by achirametavisca — APPROVED** [review]
Looks good

File: src/OnlineRetailPortal.Contracts/Models/Money.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    {
+        public double Amount { get; set; }
+        public string Currency { get; set; }
+
+        public Money(double amount, string currency)
+        {
+            this.Amount = amount;
+            this.Currency = currency;
+        }
+        public override string ToString()
```
Comment (tavisca-swapnil) [inline src/OnlineRetailPortal.Contracts/Models/Money.cs]: Do we need this?

File: src/OnlineRetailPortal.Contracts/Models/Money.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+        {
+            return this.MemberwiseClone().ToString();
+        }
```
Comment (tavisca-swapnil) [inline src/OnlineRetailPortal.Contracts/Models/Money.cs]: Do we need this?

File: src/OnlineRetailPortal.Contracts/Models/Product.cs [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        public string Description { get; set; }
+        public Image HeroImage { get; set; }
+        public Price Price { get; set; }
+        public Category Category { get; set; }
+        public Status Status { get; set; }
+        public DateTime PostDateTime { get; set; }
+        public DateTime ExpirationDate { get; set; }
+
+
+
```
Comment (tavisca-swapnil) [inline src/OnlineRetailPortal.Contracts/Models/Product.cs]: Please remove empty lines

File: src/OnlineRetailPortal.Services/Translators/AddProductTranslator.cs [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+                Url = x.Url
+            }).ToList();
+        }
+
+        public static DateTime? ToEntity(this DateTime? purchasedDate)
+        {
+            if (purchasedDate == null)
+                return null;
+            return new Core.Product().PurchasedDate = purchasedDate;
+        }
```
Comment (tavisca-swapnil) [inline src/OnlineRetailPortal.Services/Translators/AddProductTranslator.cs]: We don't need translation for predefined variable types.

**Review by tavisca-swapnil — CHANGES_REQUESTED** [review]

File: src/OnlineRetailPortal.Contracts/Models/Money.cs (hunk 1 above)
Comment (tavisca-acharpe) [reply src/OnlineRetailPortal.Contracts/Models/Money.cs]: done

**Review by tavisca-swapnil — APPROVED** [review]

File: src/OnlineRetailPortal.Contracts/Models/Money.cs (hunk 2 above)
Comment (tavisca-acharpe) [reply src/OnlineRetailPortal.Contracts/Models/Money.cs]: done

File: src/OnlineRetailPortal.Contracts/Models/Product.cs (hunk 3 above)
Comment (tavisca-acharpe) [reply src/OnlineRetailPortal.Contracts/Models/Product.cs]: done

File: src/OnlineRetailPortal.Services/Translators/AddProductTranslator.cs (hunk 4 above)
Comment (tavisca-acharpe) [reply src/OnlineRetailPortal.Services/Translators/AddProductTranslator.cs]: done

**Pull request merged**